File: CyberPunch GA/GeneticAlgorithm/GeneticAlgorithm/Individual.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GeneticAlgorithm
{
    class Individual
    {
        static int defaultGeneLength = 64;
        private byte[] genes = new byte[defaultGeneLength];
        Random rndgen = new Random();

        private int fitness = 0;

        /// <summary>
        /// Create a random individual
        /// </summary>
        public void GenIndividual()
        {
            for (int i = 0; i < Size(); i++)
            {
                byte gene = (byte)Math.Round((float)rndgen.Next(0, 2));
                genes[i] = gene;
            }
        }

        /// <summary>
        /// Sets the default length of the genes
        /// </summary>
        /// <param name="length"></param>
        public static void SetDefaultGeneLength(int length)
        {
            defaultGeneLength = length;
        }

        /// <summary>
        /// Returns the defaultGeneLength
        /// </summary>
        /// <returns></returns>
        public static int GetDefaultGeneLength()
        {
            return defaultGeneLength;
        }

        /// <summary>
        /// Returns the gene at the given index
        /// </summary>
        /// <param name="index"></param>
        /// <returns></returns>
        public byte GetGene(int index)
        {
            return genes[index];
        }

        /// <summary>
        /// Sets the gene at the given index to
        /// the value and sets its fitness to zero
        /// </summary>
        /// <param name="index"></param>
        /// <param name="value"></param>
        public void SetGene(int index, byte value)
        {
            genes[index] = value;
            fitness = 0;
        }

        /// <summary>
        /// Returns the fitness of the gene
        /// </summary>
        /// <returns></returns>
        public int GetFitness()
        {
            if(fitness == 0)
            {
                fitness = FitnessCalculator.GetFitness(this);
            }
            return fitness;
        }


        /// <summary>
        /// Return the length of the genes[]
        /// </summary>
        /// <returns></returns>
        public int Size()
        {
            return genes.Length;
        }

        /// <summary>
        /// Override ToString function to
        /// work with genes
        /// </summary>
        /// <returns></returns>
        public override string ToString()
        {
            String geneStr = "";
            for (int i = 0; i < Size(); i++)
            {
                geneStr += GetGene(i);
            }
            return geneStr;
        }
    }
}

File: TrexANN2/ml-agents-0.7.0/ml-agents/mlagents/trainers/GA/Individual.py

from random import random

class Individual:

    def __init__(self):
        self.defaultGeneLength = 64
        self.genes = []
        self.fitness = 0

    def size(self):
        return len(self.genes)

    def gen_individual(self):
        for i in range(size()):
            gene = random.range(0, 2)
            self.genes[i] = gene

    def set_default_gene_length(self, length):
        self.defaultGeneLength = length

    def get_default_gene_length(self):
        return self.defaultGeneLength

    def get_gene(self, index):
        return self.genes[index]

    def set_gene(self, index, value):
        self.genes[index] = value
        self.fitness = 0

    def get_fitness(self):
        if self.fitness == 0:
            #self.fitness = FitnessCalculator.get_fitness(self)
        return self.fitness